### ISSUE TITLE
Compatibility with ESLint 7.0.0

### ISSUE THREAD

**Issue by mradionov** [mradionov/eslint-plugin-disable#28]
https://eslint.org/blog/2020/05/eslint-v7.0.0-released

Check if nothing breaks and if something can be improved (like using async API instead of CLIEngine)

**Comment by gajus** [comment]
It does not seem to work with v7.0.0.

At the very least, the rules are still executed, but their errors are ignored.

Here is our config:

```json
{
  "extends": [
    "canonical"
  ],
  "plugins": [
    "disable"
  ],
  "processor": "disable/disable",
  "root": true,
  "settings": {
    "disable/plugins": "*"
  }
}

```

and different plugins are still being executed:

```
Rule                                            | Time (ms) | Relative
:-----------------------------------------------|----------:|--------:
@typescript-eslint/no-confusing-void-expression |  5083.679 |    19.1%
import/namespace                                |  3880.008 |    14.6%
import/no-cycle                                 |  2375.717 |     8.9%
import/extensions                               |  1684.163 |     6.3%
import/no-extraneous-dependencies               |  1094.319 |     4.1%
import/no-deprecated                            |   869.672 |     3.3%
import/no-unresolved                            |   861.904 |     3.2%
import/order                                    |   719.915 |     2.7%
import/no-duplicates                            |   665.253 |     2.5%
import/no-self-import                           |   653.202 |     2.5%

```

**Comment by mradionov** [comment]
Thanks, I'll check it out.

My plugin is designed in a way that all the rules are indeed being executed, I just don't report the errors for plugins that are "disabled". I understand that it might not be the expected behavior, but I was not able to find any other way for a plugin to work using the current ESLint API. Maybe something has changed in 7.x, I'll take a look.

**Comment by gajus** [comment]
I took a slightly different approach:

https://github.com/eslint/eslint/issues/14026#issuecomment-765678963

This seems to work fine.

**Comment by mradionov** [comment]
That's an interesting idea. I tried to follow it in my plugin without breaking anything by injecting `/* eslint-disable ... */` into each file using preprocessors. This comment would include all the rules for all disabled plugins, and it would be injected per file, but unfortunately, ESLint parses the source files first (and runs the rules), and only after that, it looks into `/* eslint-disable */` comments. So in a way core ESLint has a similar issue - it runs all the rules and checks inline comments afterwards. This was the first idea which I came up with once I saw your solution, maybe I'll think of something else later.

In regards to upgrading the plugin to ESLint 7.x API - not much has changed, the plugin still works. Using new API does not improve plugin performance according to my tests. The upgrade is not necessary at the moment, but I might do it anyways.

**Comment by jdanil** [comment]
It looks like `eslint-plugin-disable` no longer works in `eslint` v8 as `CLIEngine` has now been dropped.

The breaking changes don't look massive. It seems that instead of `CLIEngine`, `EngineLoader.js` should instead use `ESLint` and instead of calling `getConfigForFile` `Plugin.js` should call `calculateConfigForFile`. The problem is that `calculateConfigForFile` returns a promise, and I'm not quite sure how to make it work as `eslint` does not expect `preprocess` to be asynchronous.

**Comment by mradionov** [comment]
Thanks, I'll take a look

**Comment by gajus** [comment]
For what it is worth, https://www.npmjs.com/package/eslint-disable still works as expected.

The benefit of `eslint-disable` over this plugin is that it actually disables rules / plugins, whereas this plugin only hides errors from those plugins / rules.

**Comment by jdanil** [comment]
Thanks @gajus 👍 Wasn't aware of `eslint-disable`.

**Comment by mradionov** [comment]
Well, as you've said, this `calculateConfigForFile` indeed returns a Promise, but the processor API is still synchronous. `calculateConfigForFile` does not yet have any async stuff inside of it, it just calls the old CLI engine, so as a quick temporary fix I've decided to just use this old CLI engine until I figure out how to fix it properly. This fix should be available in version 2.0.2.

**Comment by jdanil** [comment]
Thanks for the quick fix @mradionov 🙏 Didn't realise `CLIEngine` could still be used in v8.
